### ISSUE TITLE
It takes a long time to make the first request

### ISSUE THREAD

**Issue by kaatt** [szmarczak/http2-wrapper#38]
With got v11:

```
> console.time('done');got('https://facebook.com').then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 302.025ms
> console.time('done');got('https://facebook.com', { http2: true }).then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 60188.968ms

> console.time('done');got('https://twitter.com').then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 2343.526ms

> console.time('done');got('https://twitter.com', { http2: true }).then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 5333.237ms

> console.time('done');got('https://google.com').then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 983.180ms
> console.time('done');got('https://google.com', { http2: true }).then(x => console.timeEnd('done'))
Promise { <pending> }
> done: 1520.592ms
```

Not sure what's causing it. Pages are speedy when I load them in Chrome.

**Comment by szmarczak** [comment]
Try

```
got('https://google.com', { http2: true }).then(x => {
    console.time('done');got('https://google.com', { http2: true }).then(x => console.timeEnd('done'))
})
```

When you make a first HTTP2 request, it may use additional time to set up the HTTP2 session. Further requests should be faster.

**Comment by kaatt** [comment]
Is there any fix for this? Facebook.com takes 60s for the first request.

```
got('https://facebook.com', { http2: true }).then(x => {
    console.time('done');got('https://facebook.com', { http2: true }).then(x => console.timeEnd('done'))
})
```

**Comment by szmarczak** [comment]
I'm working on it right now

**Comment by szmarczak** [comment]
Indeed. Fixed a few bugs, and for `google.com` (150ms compared to 360ms) it is >2x faster than HTTP1. But for `facebook.com` it waits **always** 60s for some reason.

**Comment by szmarczak** [comment]
Sometimes it gives the proper result, sometimes it waits 60s...

**Comment by kaatt** [comment]
Try changing user agent if it helps? FB could be doing some sniffing on their end

**Comment by szmarczak** [comment]
Nope.

```js
	http2wrapper.request('https://facebook.com', () => {
		console.log('got response');
	}).end();
```

this works as expected... but it doesn't work with Got for some reason...

**Comment by szmarczak** [comment]
It has to do something with redirects or HTTP2 origins. If you add `followRedirect: false` it works as expected.

**Comment by szmarczak** [comment]
No. If you query `https://www.facebook.com` and disable redirects it also hangs.

**Comment by szmarczak** [comment]
Hmmm.

```js
http2wrapper.request('https://www.facebook.com', () => {
	console.log('got response');
}).end();
```

works as expected so I suspect it's a bug in Got.

**Comment by kaatt** [comment]
Still possible (although unlikely) that facebook is UA sniffing and when it sees the got user-agent it rate limits or throttles.

**Comment by szmarczak** [comment]
no

**Comment by szmarczak** [comment]
`http2: true, decompress: false` works as expected but if you set `decompress` to true it hangs

**Comment by szmarczak** [comment]
Testing on `http2-wrapper@master`

**Comment by kaatt** [comment]
you can maybe try other popular sites and see if the same issue is happening with any of them. might help in filtering out the issue with compression

**Comment by szmarczak** [comment]
```js
request('https://www.facebook.com', {
	headers: {
		'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.70 Safari/537.36',
		'accept-encoding': 'gzip, deflate, br'
	}
}, response => {
	response.resume();
	response.on('data', chunk => {
		console.log(chunk.toString().length);
	});
	response.once('end', () => {
		console.log('got end');
	});
}).end();
```

It's a `http2-wrapper` bug 10000%

**Comment by szmarczak** [comment]
Either it's a bug in Node.js, nghttp2 (the underlying HTTP2 implementation) or in Facebook itself:

```js
const {connect} = require('http2');
const session = connect('https://www.facebook.com');
session.once('remoteSettings', () => {
	console.log('got settings');
	session.request({
		'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.70 Safari/537.36',
		'accept-encoding': 'gzip, deflate, br'
	}).once('response', () => {
		console.log('got headers');
	}).on('data', chunk => {
		console.log(chunk.toString().length);
	}).on('end', () => {
		console.log('got end');
	}).resume();
});
```

**Comment by szmarczak** [comment]
It's a bug in Facebook, if you remove `, br` from `accept-encoding` you receive the `end` event.

**Comment by szmarczak** [comment]
It may be also a bug in Node.js, I need to check that.

**Comment by kaatt** [comment]
could this be an issue in proxygen? https://engineering.fb.com/production-engineering/introducing-proxygen-facebook-s-c-http-framework/

i see chrome's only sending "br" for "accept-encoding"

**Comment by szmarczak** [comment]
that's what I'm sending too

**Comment by szmarczak** [comment]
It's a Node.js bug, `curl --http2 -H 'accept-encoding: gzip, deflate, br' https://www.facebook.com>/dev/null` works as expected.

**Comment by szmarczak** [comment]
Please see the issue above. I'll release a new beta right now.

**Comment by szmarczak** [comment]
Released `1.0.0-beta.4.4`

**Comment by kaatt** [comment]
thanks!
